src/views/common/WithdrawDash.tsx

```diff
@@ -0,0 +1,198 @@
+import { ModalIcon } from "../../utils/icons";
+import { TokenIcon } from "../../utils/icons";
+import { forEach } from "lodash";
+import { string } from "prop-types";
+
+type WithdrawDashProps = {
+	token: ReserveTokenDefinition;
+};
+
+const WithdrawDash: React.FC<WithdrawDashProps> = ({
+	token,
+}) => {
+	const { account: userAccountAddress } = useAppWeb3();
+	const { data: reserves } = useAllReserveTokensWithData();
+	const reserve = React.useMemo(
+		() =>
+			reserves?.find(reserve => reserve.tokenAddress === token.tokenAddress) ??
+			reserves?.find(
+				reserve =>
+					reserve.tokenAddress.toLowerCase() ===
+					token.tokenAddress.toLowerCase()
+			),
+		[reserves, token.tokenAddress]
+	);
+	const { data: reserveProtocolData } = useProtocolReserveData(reserve?.tokenAddress);
+	const { data: reserveConfiguration } = useProtocolReserveConfiguration(reserve?.tokenAddress);
+	const { data: userAccountData } = useUserAccountData(userAccountAddress ?? undefined);
+	const { data: allReservesData } = useUserReserveAssetBalancesDaiWei();
+	const { data: tokenBalance } = useUserAssetBalance(token.tokenAddress);
+	const { data: aTokenBalance } = useUserAssetBalance(reserve?.aTokenAddress);
+	const { data: utilizationData } = useAssetUtilizationRate(token.tokenAddress);
+	const { data: assetPriceInDai } = useAssetPriceInDai(reserve?.tokenAddress);
+	const utilizationRate = utilizationData?.utilizationRate;
+	const liquidityAvailable = reserveProtocolData?.availableLiquidity;
+	const isCollateralized = reserveConfiguration?.usageAsCollateralEnabled;
+	const maximumLtv = reserveConfiguration?.ltv;
+	const currentLtv = userAccountData?.currentLtv;
+	const variableDepositAPY = reserveProtocolData?.variableBorrowRate;
+	const healthFactor = userAccountData?.healthFactor;
+
+	const totalCollateralValue = React.useMemo(() => {
+		return allReservesData?.reduce(
+			(memo: BigNumber, next) =>
+				next.daiWeiPriceTotal !== null ? memo.add(next.daiWeiPriceTotal) : memo,
+			constants.Zero
+		);
+	}, [allReservesData])
+
+	const collateralComposition = React.useMemo(() => {
+		const compositionArray = allReservesData?.map(next => {
+			if (next.daiWeiPriceTotal != undefined && next.decimals != undefined && totalCollateralValue != undefined) {
+				const decimalPower = BigNumber.from(10).pow(next.decimals);
+				return next.daiWeiPriceTotal.mul(decimalPower).div(totalCollateralValue)
+			}
+			else return BigNumber.from(0);
+		});
+		return (compositionArray) ? compositionArray.map(share => {
+			if (share.gt(0)) {
+				return formatEther(share.mul(100))
+			}
+			else return null
+		}) : []
+	}, [allReservesData, totalCollateralValue]);
+
+	const collateralData = collateralComposition.map((x, index) => {
+		if (x != null) return x.substr(0, x.indexOf(".") + 3);
+	})
+
+	const [isSmallerThan900] = useMediaQuery("(max-width: 900px)");
+	const [isSmallerThan400] = useMediaQuery("(max-width: 400px)");
```

Comment:
Less hooks is more efficient, do so when they can be combined.
```suggestion
	const [isSmallerThan400, isSmallerThan900] = useMediaQuery(["(max-width: 400px)", "(max-width: 900px)"]);
```

Reply:
ok fixed